kibana-reports/server/routes/savedSearchReportHelper.ts

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import { v1 as uuidv1 } from 'uuid';
+import {
+  buildQuery,
+  convertToCSV,
+  getEsData,
+  getSelectedFields,
+  metaData,
+} from './utils/dataReportHelpers';
+import {
+  IClusterClient,
+  IScopedClusterClient,
+} from '../../../../src/core/server';
+
+/**
+ * Specify how long scroll context should be maintained for scrolled search
+ */
+const scrollTimeout = '1m';
+
+export async function createSavedSearchReport(
+  report: any,
+  client: IClusterClient | IScopedClusterClient
+) {
+  await populateMetaData(client, report);
+  const data = await generateCsvData(client);
+
+  const timeCreated = new Date().toISOString();
+  const fileName = getFileName() + '.csv';
+  return {
+    timeCreated,
+    dataUrl: data,
+    fileName,
+  };
+
+  function getFileName(): string {
+    return `${
+      report.report_definition.report_params.report_name
+    }_${timeCreated}_${uuidv1()}`;
+  }
+}
+
+/**
+ * Populate parameters and saved search info related to meta data object.
+ * @param client  ES client
+ * @param report  Report input
+ */
+async function populateMetaData(
+  client: IClusterClient | IScopedClusterClient,
+  report: any
+) {
+  metaData.saved_search_id =
+    report.report_definition.report_params.core_params.saved_search_id;
+  metaData.report_format =
+    report.report_definition.report_params.core_params.report_format;
+  metaData.start = report.time_from;
+  metaData.end = report.time_to;
+
+  // Get saved search info
+  let resIndexPattern: any = {};
+  const ssParams = {
+    index: '.kibana',
+    id: 'search:' + metaData.saved_search_id,
+  };
+  const ssInfos = await client.callAsInternalUser('get', ssParams);
```

Comment:
change all ES calls to use this function. `callCluster`
https://github.com/opendistro-for-elasticsearch/kibana-reports/blob/63937d409d564d8756af3b1a84e86d67bd37e675/kibana-reports/server/routes/utils/reportHelper.ts#L136
https://github.com/opendistro-for-elasticsearch/kibana-reports/blob/63937d409d564d8756af3b1a84e86d67bd37e675/kibana-reports/server/routes/utils/reportHelper.ts#L193

Reply:
Sure, will do. Btw, I'm think moving this function and `getFileName` to a new util file. Otherwise I have to either copy or call it here which I assume will cause dependency cycle issue?

Reply:
Sounds good, please move that function to new file.